src/main/java/com/vladyslavpalamarchuk/trackmycoin/service/MonitoringService.java

```diff
@@ -0,0 +1,127 @@
+package com.vladyslavpalamarchuk.trackmycoin.service;
+
+import com.vladyslavpalamarchuk.trackmycoin.adaptors.api.BinanceApiClient;
+import com.vladyslavpalamarchuk.trackmycoin.adaptors.persistence.MonitoringRepository;
+import com.vladyslavpalamarchuk.trackmycoin.adaptors.persistence.UserRepository;
+import com.vladyslavpalamarchuk.trackmycoin.adaptors.telegram.TelegramBotClient;
+import com.vladyslavpalamarchuk.trackmycoin.domain.Monitoring;
+import com.vladyslavpalamarchuk.trackmycoin.domain.User;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Getter
+@Slf4j
+public class MonitoringService {
+
+  private final MonitoringRepository monitoringRepository;
+  private final TelegramBotClient telegramBotClient;
+  private final UserRepository userRepository;
+  private final BinanceApiClient binanceApiClient;
+  private final NotificationService notificationService;
+  private final String USER_PREFIX = "User_";
+  private final String TICKER_CURRENCY = "USDT";
+  private final BigDecimal DECIMAL_FORMAT_TYPE = BigDecimal.valueOf(1);
+
+  public void add(String ticker, String price, long chatId) {
+    Monitoring monitoring = new Monitoring();
+    monitoring.setTicker(ticker.toUpperCase() + TICKER_CURRENCY);
+
+    BigDecimal targetPrice = new BigDecimal(price);
+    monitoring.setTargetPrice(targetPrice);
+
+    User user = userRepository.findUserByChatId(chatId);
+    monitoring.setUser(user);
+    monitoring.setCreatedBy(USER_PREFIX + chatId);
+    monitoring.setUpdatedBy(USER_PREFIX + chatId);
+    monitoringRepository.save(monitoring);
+
+    telegramBotClient.sendMessage(chatId, "The coin is available! Monitoring successfully added.");
+  }
+
+  public String get(Long userId) {
```

Comment:
You missed single responsibility here. This method should just retrieve monitorings looks like but he is finding user, sending message to TG